nuxeo-services/nuxeo-platform-web-common/src/main/resources/web/nuxeo.war/login.jsp

```diff
@@ -524,6 +524,16 @@ input:-webkit-autofill:focus {
 </div>
 
 <script type="text/javascript">
+  // Handle requested URL for Web UI
+  var webUIPrefix = 'ui'
+  var requestedURL = document.getElementById('requestedUrl').value;
+  if (!requestedURL || requestedURL.startsWith(webUIPrefix)) {
+    var indexOfHash = window.location.href.indexOf('#!');
+    if (indexOfHash > -1) {
+      document.getElementById('requestedUrl').value = webUIPrefix + '/' + window.location.href.substring(indexOfHash);
+    }
+  }
+
```

Comment:
In fact the issue is not on a requestedUrl containing #!, but the fact that the #! part of a Web UI URL isn't sent to the server, so when asking for a Web UI URL such as `http://localhost:8080/nuxeo/ui/#!/browse/default-domain/workspaces/Test WS/Test MD Note` the requestedUrl would never contain the #!/... part, thus the initial bug.

That's why we came up with this solution of parsing the browser's URL in JS to add this missing part to the requestedUrl, this only in the case of a Web UI URL.